sktime/classifiers/shapelet_based/mrseql/mrseql.pyx

```diff
@@ -0,0 +1,439 @@
+    # change arbitrary binary labels to -1, 1 labels as SEQL can only work with -1, 1
+    def __to_tmp_labels(self, y):
+        return [1 if l == self.classes_[1] else -1 for l in y]
+
+
+    def __transform_time_series(self, ts_x):
+        multi_tssr = []
+
+        # generate configuration if not predefined
+        if not self.config:
+            min_ws = 16
+            max_ws = ts_x.shape[1]
+            pars = [[w, 16, 4] for w in range(min_ws, max_ws, int(np.sqrt(max_ws)))]
+
+            if 'sax' in self.symbolic_methods:
+                for p in pars:
+                    self.config.append({'method':'sax','window':p[0],'word':p[1],'alphabet':p[2]})
+
+            if 'sfa' in self.symbolic_methods:
+                for p in pars:
+                    self.config.append({'method':'sfa','window':p[0],'word':8,'alphabet':p[2]})
+
+
+        for cfg in self.config:
+
+            tssr = []
+
+            if cfg['method'] == 'sax': # convert time series to SAX
+                ps = PySAX(cfg['window'],cfg['word'],cfg['alphabet'])
+                for ts in ts_x:
+                    sr = ps.timeseries2SAXseq(ts)
+                    tssr.append(sr)
+
+            if cfg['method'] == 'sfa':  # convert time series to SFA
+                if (cfg['window'],cfg['word'],cfg['alphabet']) not in self.sfas:
+                    sfa = AdaptedSFA(cfg['window'],cfg['word'],cfg['alphabet'])
+                    sfa.fit(ts_x)
+                    self.sfas[(cfg['window'],cfg['word'],cfg['alphabet'])] = sfa
+                for ts in ts_x:
+                    sr = self.sfas[(cfg['window'],cfg['word'],cfg['alphabet'])].timeseries2SFAseq(ts)
+                    tssr.append(sr)
+
+            multi_tssr.append(tssr)
+
+
+        return multi_tssr
+
+
+    def __fit_binary_problem(self, mr_seqs, labels):
+        models = []
+        for rep in mr_seqs:
+            m = PySEQL()
+            m.learn(rep, labels)
+            models.append(m)
+        return models
+
+    def __fit_multiclass_problem(self, mr_seqs, labels):
+        models = []
+        for rep in mr_seqs:
+            m = OVASEQL(self.classes_)
+            m.learn(rep, labels)
+            models.append(m)
+        return models
+
+
+
+    # represent data (in multiple reps form) in feature space
+    def __to_feature_space(self, mr_seqs):
+        full_fm = []
+
+        for rep, model in zip(mr_seqs, self.seql_models):
+            seq_features = model.get_sequence_features(False)
+            # print(seq_features)
+            fm = np.zeros((len(rep), len(seq_features)))
+            for i,s in enumerate(rep):
+                for j,f in enumerate(seq_features):
+                    if f in s:
+                        fm[i,j] = 1
+            full_fm.append(fm)
+
+
+        full_fm = np.hstack(full_fm)
+        return full_fm
+
+    '''
+    Check if X input is correct.
+    From dictionary_based/boss.py
+    '''
+    def __X_check(self,X):
+        if isinstance(X, pd.DataFrame):
```

Comment:
Please also use our input validation methods from [here](https://github.com/alan-turing-institute/sktime/blob/master/sktime/utils/validation/supervised.py), i.e. `validate_X` and `check_is_univariate`